pkg/controller/directvolumemigrationprogress/directvolumemigrationprogress_controller.go

```diff
@@ -247,14 +248,25 @@ func (r *ReconcileDirectVolumeMigrationProgress) reportContainerStatus(pvProgres
 	}
 
 	if containerStatus == nil {
-		pvProgress.Status.SetCondition(migapi.Condition{
-			Type:     InvalidPod,
-			Status:   migapi.True,
-			Reason:   NotFound,
-			Category: migapi.Critical,
-			Message: fmt.Sprintf("The spec.podRef %s must reference a `Pod` with container name %s",
-				path.Join(podRef.Namespace, podRef.Name), containerName),
-		})
+		if pod.Status.Phase == kapi.PodFailed {
+			pvProgress.Status.PodPhase = kapi.PodFailed
+			pvProgress.Status.SetCondition(migapi.Condition{
+				Type:     PodFailed,
+				Status:   migapi.True,
+				Reason:   pod.Status.Reason,
+				Category: migapi.Warn,
+				Message:  fmt.Sprintf("The rsync pod %s is in %s state because of the following reason: %s", path.Join(pod.Name, pod.Namespace), pod.Status.Phase, pod.Status.Message),
```

Comment:
adding to Alay's question, since this `if` block will be only triggered when Pod phase is `kapi.PodFailed`, it doesnt make sense to me why we have a dynamic message here. This could as well be a static message saying - we found out that the pod is failed. There could also be scenarios where Status.Message is not set, in that case the message here will add to confusion

Reply:
Nice catch! removing adding critical condition altogether as per alay's remark